vcd/resource_vcd_org.go

```diff
@@ -171,51 +182,119 @@ func resourceOrgUpdate(d *schema.ResourceData, m interface{}) error {
-	oldOrgFullName := oldOrgFullNameRaw.(string)
-	newOrgFullName := newOrgFullNameRaw.(string)
-
-	if !strings.EqualFold(oldOrgFullName, newOrgFullName) {
-		return fmt.Errorf("__ERROR__ Not Updating org_full_name , API NOT IMPLEMENTED !!!!")
+	orgName, fullName, err := getOrgNames(d)
+	if err != nil {
+		return err
 	}
+	log.Printf("Reading Org with id %s", d.State().ID)
 
-	log.Printf("Reading Org WIth id %s", d.State().ID)
-
-	org, err := govcd.GetAdminOrgByName(vcdClient.VCDClient, d.Get("name").(string))
-
-	if err != nil || org == (govcd.AdminOrg{}) {
-		return fmt.Errorf("error fetching Org: %s", d.Get("name").(string))
+	adminOrg, err := vcdClient.VCDClient.GetAdminOrgByName(orgName)
+	if err != nil {
+		return fmt.Errorf("error fetching Org: %s", orgName)
 	}
 
 	settings := getSettings(d)
-	org.AdminOrg.Name = orgName
-	org.AdminOrg.OrgSettings.OrgGeneralSettings = settings.OrgGeneralSettings
+	adminOrg.AdminOrg.Name = orgName
+	adminOrg.AdminOrg.FullName = fullName
+	adminOrg.AdminOrg.Description = d.Get("description").(string)
+	adminOrg.AdminOrg.IsEnabled = d.Get("is_enabled").(bool)
+	adminOrg.AdminOrg.OrgSettings.OrgGeneralSettings = settings.OrgGeneralSettings
 
 	log.Printf("Org with id %s found", d.State().ID)
-	_, err = org.Update()
+	task, err := adminOrg.Update()
 
 	if err != nil {
-		log.Printf("Error updating Org with id %s : %#v", d.State().ID, err)
-		return fmt.Errorf("error updating Org %#v", err)
+		log.Printf("Error updating Org with id %s : %s", d.State().ID, err)
+		return fmt.Errorf("error updating Org %s", err)
+	}
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		log.Printf("Error completing update of Org with id %s : %s", d.State().ID, err)
+		return fmt.Errorf("error completing update of Org %s", err)
 	}
 
 	log.Printf("Org with id %s updated", d.State().ID)
 	return nil
 }
 
+func setOrgData(d *schema.ResourceData, adminOrg *govcd.AdminOrg) error {
+	err := d.Set("name", adminOrg.AdminOrg.Name)
+	if err != nil {
+		return err
+	}
+	err = d.Set("full_name", adminOrg.AdminOrg.FullName)
+	if err != nil {
+		return err
+	}
+	err = d.Set("description", adminOrg.AdminOrg.Description)
+	if err != nil {
+		return err
+	}
+	err = d.Set("is_enabled", adminOrg.AdminOrg.IsEnabled)
+	if err != nil {
+		return err
+	}
+	err = d.Set("deployed_vm_quota", adminOrg.AdminOrg.OrgSettings.OrgGeneralSettings.DeployedVMQuota)
+	if err != nil {
+		return err
+	}
+	err = d.Set("stored_vm_quota", adminOrg.AdminOrg.OrgSettings.OrgGeneralSettings.StoredVMQuota)
+	if err != nil {
+		return err
+	}
+	err = d.Set("can_publish_catalogs", adminOrg.AdminOrg.OrgSettings.OrgGeneralSettings.CanPublishCatalogs)
+
+	return err
+}
+
+// Retrieves an Org resource from vCD
 func resourceOrgRead(d *schema.ResourceData, m interface{}) error {
 	vcdClient := m.(*VCDClient)
 
-	log.Printf("Reading Org with id %s", d.State().ID)
-	org, err := govcd.GetAdminOrgByName(vcdClient.VCDClient, d.Get("name").(string))
+	identifier := d.State().ID
```

Comment:
Wondering if there is any difference between `d.Id()` we regularly use.

Reply:
there isn't. I changed to using it. I actually cleaned up the whole function